src/models/room.ts

```diff
@@ -809,7 +809,7 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
 
         const lastThreadEvent = lastThread.events[lastThread.events.length - 1];
 
-        return (lastRoomEvent?.getTs() ?? 0) > (lastThreadEvent.getTs() ?? 0) ? lastRoomEvent : lastThreadEvent;
+        return (lastRoomEvent?.getTs() ?? 0) > (lastThreadEvent?.getTs() ?? 0) ? lastRoomEvent : lastThreadEvent;
```

Comment:
Can you explain how this fixes the issue? I'm not really following.

Reply:
oh, because the issue was caused by `Cannot read properties of undefined`. Ignore me, I'm being stupid.

That said: I would expect to see a test for this situation, and I don't?

Reply:
I encountered the same problem as Karl this morning. Console showed `Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'getTs')`.

Reply:
> oh, because the issue was caused by `Cannot read properties of undefined`. Ignore me, I'm being stupid.
> 
> That said: I would expect to see a test for this situation, and I don't?

The test is to create a thread anyway (even it is empty) : https://github.com/matrix-org/matrix-js-sdk/pull/3393/files#diff-711c041decd50c6513b88182abb538a73c5c10148570cce573b486b876359606R177 In this case it runs into the `Cannot read properties of undefined` error.

Reply:
I don't really see how that works. `mkThread` adds two events to the thread, afaict, so we're still not testing the situation where `lastThread.events` is empty?